src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -799,4 +799,47 @@ public function getTestsForOctalNumbers()
             'negative octal number' => [-28, '-034'],
         ];
     }
+
+    /**
+     * @dataProvider phpObjectTagWithEmptyValueProvider
+     */
+    public function testPhpObjectWithEmptyValue($expected, $value)
+    {
+        $this->assertSame($expected, Inline::parse($value, Yaml::PARSE_OBJECT));
+    }
+
+    public function phpObjectTagWithEmptyValueProvider()
+    {
+        return [
+            [false, '!php/object'],
+            [false, '!php/object '],
+            [false, '!php/object  '],
+            [[false], '[!php/object]'],
+            [[false], '[!php/object ]'],
+            [[false, 'foo'], '[!php/object  , foo]'],
+        ];
+    }
+
+    /**
+     * @dataProvider phpConstTagWithEmptyValueProvider
+     */
+    public function testPhpConstTagWithEmptyValue($expected, $value)
+    {
+        $this->assertSame($expected, Inline::parse($value, Yaml::PARSE_CONSTANT));
+    }
+
+    public function phpConstTagWithEmptyValueProvider()
+    {
+        return [
+            ['', '!php/const'],
+            ['', '!php/const '],
+            ['', '!php/const  '],
```

Comment:
previously it would be `" "` isnt it? :P

Reply:
i mean, if `constant(' ')` were to be defined, the key would be `"some"`

Reply:
never mind :D it's still OK if it would be defined.

Reply:
It would work with no error reporting on notices.

Reply:
Anyway that's such a weird case. I think it doesn't matter.

Reply:
> never mind :D it's still OK if it would be defined.

No. My previous replies were to your first message. If the `''` constant is defined, it is never resolved now.

Reply:
`return defined('') ? constant('') : '';`? :joy:

Reply:
Yes we can stay backward compatible by checking it. But do we want to? 😅

Reply:
no, but that's for 5.x. It would make the feature removal/deprecation more explicit.